tests/geometry/test-aabb.cpp

```diff
@@ -0,0 +1,312 @@
+        create_box_vector< dimension >( nb_boxes, box_size );
+    AABBTree< dimension > aabb( box_vector );
+
+    BoxAABBEvalIntersection< dimension > eval_intersection =
+        BoxAABBEvalIntersection< dimension >( box_vector );
+
+    for( geode::index_t i : Range( nb_boxes - 1 ) )
+    {
+        for( index_t j : Range( nb_boxes - 1 ) )
+        {
+            Point< dimension > query;
+            // query boxes will be at internal corner grid
+            query.set_value( 0, (double) i + box_size );
+            query.set_value( 1, (double) j + box_size );
+            BoundingBox< dimension > box_query =
+                create_bounding_box( query, box_size );
+
+            eval_intersection.box_intersections_.clear();
+            aabb.compute_bbox_element_bbox_intersections(
+                box_query, eval_intersection );
+
+            OPENGEODE_EXCEPTION(
+                eval_intersection.box_intersections_.size() == 4,
+                "[Test] Error ... wrong number of intersected boxes" );
+
+            std::set< index_t > expected_set;
+            expected_set.emplace( global_box_index( i, j, nb_boxes ) );
+            expected_set.emplace( global_box_index( i, j + 1, nb_boxes ) );
+            expected_set.emplace( global_box_index( i + 1, j, nb_boxes ) );
+            expected_set.emplace( global_box_index( i + 1, j + 1, nb_boxes ) );
+            OPENGEODE_EXCEPTION(
+                eval_intersection.box_intersections_ == expected_set,
+                "[Test] Error ... wrong intersected set of box" );
+        }
+    }
+    Point< dimension > query;
+    // query box will be at top right corner grid
+    query.set_value( 0, (double) nb_boxes - 1 + box_size );
+    query.set_value( 1, (double) nb_boxes - 1 + box_size );
+    BoundingBox< dimension > box_query = create_bounding_box( query, box_size );
+
+    eval_intersection.box_intersections_.clear();
+    aabb.compute_bbox_element_bbox_intersections(
+        box_query, eval_intersection );
+
+    OPENGEODE_EXCEPTION( eval_intersection.box_intersections_.size() == 1,
+        "[Test] Error ... wrong number of intersected boxes" );
+
+    std::set< index_t > expected_set;
+    expected_set.emplace(
+        global_box_index( nb_boxes - 1, nb_boxes - 1, nb_boxes ) );
+    OPENGEODE_EXCEPTION( eval_intersection.box_intersections_ == expected_set,
+        "[Test] Error ... wrong intersected set of box" );
+}
+
+template < index_t dimension >
+void test_self_intersections()
+{
+    index_t nb_boxes = 10;
+    // create a grid of intersected boxes
+    std::vector< BoundingBox< dimension > > box_vector =
+        create_box_vector< dimension >( nb_boxes, 0.75 );
+    // create a grid of tangent boxes included in previous boxes
+    std::vector< BoundingBox< dimension > > box_vector2 =
+        create_box_vector< dimension >( nb_boxes, 0.50 );
+    box_vector.insert(
+        box_vector.end(), box_vector2.begin(), box_vector2.end() );
```

Comment:
```suggestion
    const index_t nb_boxes{ 10 };
    // Create a grid of intersecting boxes
    auto box_vector =
        create_box_vector< dimension >( nb_boxes, 0.75 );
    // Create a grid of tangent boxes included in previous boxes
    const auto box_vector2 =
        create_box_vector< dimension >( nb_boxes, 0.50 );
    box_vector.insert(
        box_vector.end(), box_vector2.begin(), box_vector2.end() );
```
Maybe absl provides a function `c_insert`

Reply:
I coud not find this function.

Reply:
Ok